metaserver/src/main/scala/scala/meta/languageserver/ScalametaLanguageServer.scala

```diff
@@ -103,19 +90,19 @@ class ScalametaLanguageServer(
       MulticastStrategy.Publish,
       OverflowStrategy.DropOld(2)
     )
+  val (configurationSubscriber, configurationPublisher) =
+    ScalametaLanguageServer.configurationStream(connection)
```

Comment:
Why do we have connection here as well as lspIn and lspOut passed in?  I thought connection was just a wrapper around lspIn and lspOut.

Reply:
connection has higher level methods such as `showMessage`.

`lspIn` and `lspOut` are just used to initialize it (via the `LanguageServer` constructor)